### DISCUSSION TITLE
User defined minibatching

### DISCUSSION THREAD

**Discussion by Neel-132** [pyg-team/pytorch_geometric#8803] [Q&A]
So, I have a set of graphs for link prediction. Now, the way I want to pass a set of graphs in each batch is custom. Also, while passing it to the data loader I want to sample neighbors for a particular set of nodes randomly. More clearly consider this image depicted below:
![LinkPred](https://github.com/pyg-team/pytorch_geometric/assets/71507677/f47fcf67-825e-49e7-aee1-1b0cd3b74f3a)
I want to pass the graph enclosed in the red dotted circle in one batch and while passing I want to sample the nodes associated with the CA1, CA2,... nodes (connected via violet colored edges) randomly. Is there a way to do that without writing custom dataloader?

**Comment by wsad1** [comment]
Pyg has several dataloaders you can find them [here](https://pytorch-geometric.readthedocs.io/en/latest/modules/loader.html#torch-geometric-loader). Check out LinkNeighborLoader and RandomNodeLoader. If none among the loaders fit your use case you might have to write your own dataloader.
